lib/Conversion/TorchToLinalg/DataMovement.cpp

```diff
@@ -513,20 +533,15 @@ class ConvertAtenViewOp : public OpConversionPattern<AtenViewOp> {
         if (failed(minimallyCollapseDimHelper(
                 op, rewriter, outputDim, nextUnchangedOutput, inputDim,
                 nextUnchangedInput, outputShape, inputShapeVec,
-                inputAssociations.back())))
+                outputAssociations.back(), inputAssociations.back()))) {
           return failure();
+        }
         hasDynamic = false;
         inputDim = inputAssociations.back().back() + 1;
         outputDim++;
         continue;
       }
 
-      if (inputDim != nextUnchangedInput || outputDim != nextUnchangedOutput) {
-        return rewriter.notifyMatchFailure(
-            op, "could not match input tensor shape to output shape; "
-                "potentially unsupported view shape");
-      }
```

Comment:
I think this check is still important for making sure the view op is valid (think [2, 3] -> [2, 3, 4]). The only reason it says potentially unsupported is because match failures from dynamic dims overlap here and are unimplemented rather than invalid. It would probably be worth updating the error message though. This check should be fine after properly updating the inputDim and outputDim above.